src/main/java/seedu/address/model/Model.java

```diff
@@ -109,7 +110,12 @@
      * Updates the filter of the filtered person list to filter by the given {@code predicate}.
      * @throws NullPointerException if {@code predicate} is null.
      */
-    void updateFilteredPersonList(Predicate<Person> predicate);
+    void updateFilteredPersonList(Predicate<? super Person> predicate);
+
+    /**
+     * Returns the current {@code Predicate} of the person filtered list
+     */
+    Optional<Predicate<? super Person>> getPersonPredicate();
```

Comment:
May I ask what is the significance of using <? super Person> rather than <Person> ?

Reply:
The getPredicate() method of FilteredList returns Predicate<? super T> and setPredicate() method of FilteredList accept Predicate<? super T> as argument. If just keep to Predicate<Person> I need to do type casting and need to suppress the warning. So I feel it is safer to change the signature than suppressing the warning.

Reply:
ok got it. thank you.